supercommand.go

```diff
@@ -68,6 +68,7 @@ type SuperCommandParams struct {
 	MissingCallback MissingCallback
 	Aliases         []string
 	Version         string
+	VersionDetail   interface{}
```

Comment:
It'd be good to document here what will be done with the thing that's passed for VersionDetail, just to save client code authors from having to read all the code to work it out. I can understand not wanting to make it a specific interface (since it's just some arbitrary data from this perspective), but saying something like "It will be serialised as yaml or json depending on the --format option specified." would do.